modules/shapefile/src/lib/parsers/parse-dbf.js

```diff
@@ -50,10 +52,20 @@ export function parseDBF(arrayBuffer, options) {
   dbfParser.write(arrayBuffer);
   dbfParser.end();
 
-  const {data} = dbfParser.result;
-  return data;
-}
+  // const data = parseRows(binaryReader, fields, nRecords, recordLength, textDecoder);
+
+  const {data, schema} = dbfParser.result;
 
+  switch (options.tables && options.tables.format) {
+    case 'table':
+      // TODO - parse columns
+      return {schema, rows: data};
```

Comment:
So this doesn't return `rows` _as an Arrow table_, it returns as a normal JS object with the Arrow metadata of what each key should be understood as?

Reply:
Yes - converting tables and batches to Arrow format is a bigger thing. 

Here I just want to create a schema that describes the binary format of the columns. This schema will obviously be used when actually converting the output to Arrow.